microgen/operations.py

```diff
@@ -29,68 +51,45 @@ def rotateEuler(
 
     :param obj: Object to rotate
     :param center: numpy array (x, y, z)
-    :param psi, theta, phi: Euler angles
+    :param psi: first Euler angle, in degrees
+    :param theta: first Euler angle, in degrees
+    :param phi: first Euler angle, in degrees
 
     :return object_r: Rotated object
     """
 
-    u = np.array([0.0, 0.0, 1.0])
-    u = np.array([np.cos(psi * np.pi / 180.0), np.sin(psi * np.pi / 180.0), 0.0])
-    z2 = np.array(
-        [
-            np.sin(psi * np.pi / 180.0) * np.sin(theta * np.pi / 180.0),
-            -np.sin(theta * np.pi / 180.0) * np.cos(psi * np.pi / 180.0),
-            np.cos(theta * np.pi / 180.0),
-        ]
-    )
+    center_vector = cq.Vector(*center)
 
-    object_r = obj.rotate(
-        cq.Vector(center[0], center[1], center[2]),
-        cq.Vector(center[0], center[1], center[2] + 1.0),
-        psi,
-    )
-    object_r = object_r.rotate(
-        cq.Vector(center[0], center[1], center[2]),
-        cq.Vector(center[0] + u[0], center[1] + u[1], center[2] + u[2]),
-        theta,
-    )
-    object_r = object_r.rotate(
-        cq.Vector(center[0], center[1], center[2]),
-        cq.Vector(center[0] + z2[0], center[1] + z2[1], center[2] + z2[2]),
-        phi,
-    )
-    return object_r
+    z, u, z2 = _getRotationAxes(psi, theta, phi)
+
+    obj = obj.rotate(center_vector, center_vector + cq.Vector(*z), psi)
+    obj = obj.rotate(center_vector, center_vector + cq.Vector(*u), theta)
+    obj = obj.rotate(center_vector, center_vector + cq.Vector(*z2), phi)
+    return obj
 
 
 def rotatePvEuler(
-    object: pv.PolyData,
-    center: np.ndarray,
+    obj: pv.PolyData,
+    center: Sequence[float],
     psi: float,
     theta: float,
     phi: float,
 ) -> pv.PolyData:
     """
     Rotates object according to XZX Euler angle convention
 
-    Parameters
-    ----------
-    object :
-        Object to rotate
-    center :
-        numpy array (x, y, z)
-    psi, theta, phi :
-        Euler angles
-
-    Returns
-    -------
-    object_r :
-        Rotated object
+    :param obj: Object to rotate
+    :param center: numpy array (x, y, z)
+    :param psi: first Euler angle, in degrees
+    :param theta: second Euler angle, in degrees
+    :param phi: third Euler angle, in degrees
+
+    :return: Rotated object
     """
+    z, u, z2 = _getRotationAxes(psi, theta, phi)
 
-    u = (np.cos(psi), np.sin(psi), 0.0)
-    z2 = (np.sin(psi) * np.sin(theta), -np.sin(theta) * np.cos(psi), np.cos(theta))
-    object_r = object.rotate_vector(
-        vector=(0, 0, 1), angle=psi, point=tuple(center), inplace=False
+    object_r = obj.rotate_vector(
```

Comment:
rotated_object couldn't be more explicit as variable name than object_r ?

Reply:
Indeed!

Reply:
I renamed it to `rotated_obj`, for consistency with the input `obj`.